Feat/413 유저/기업 홈 수정

ohofront opened the pull request:
## Related issue 🛠

[//]: # "해당하는 이슈 번호 달아주기"

- closed #413 

## Work Description ✏️

[//]: # "작업 내용 간단 소개"

1. 전체 흐름 및 요구사항
  - 홈 화면 Job / Career / Resume 카드 디자인 통일
  - 로그인한 유저 타입(account_type)에 따라 다른 카드 노출
     - 유저(UserType.USER)
         - Trending Job Lists for you (스크랩 수 기준 인기)
         - Trending Career Lists for you (스크랩 수 기준 인기)
         - Bookmarked Job
         - Bookmarked Career
     - 고용주(UserType.OWNER)
         - 요즘 인기있는 인재 (Resume 리스트)
         - 요즘 인기있는 공고 (Job 리스트)
2. 컴포넌트 분리 및 통일
 - Job 공고 카드 (HomePostCard)
       - 대표 이미지 (representative_img_url)를 backgroundImage로 적용
       - 고정 width, height (w-[9.063rem] h-[6.75rem])
       - 제목, 회사명, 주소, employment_type/visa 태그 표시
 - Career 공고 카드 (HomeCareerPostCard)
       - 기존 Job 카드 디자인과 완전히 동일하게 통일
       - careerData.img_urls[0] 이미지 사용 (없을 경우 No Image 박스 노출)
       - organizer_name, host_name 표시
       - career_category, visa 태그 표시
 - Resume 카드 (고용주 전용) (EmployerEmployeeCardList + EmployerEmployeeCard)
     - 디자인 통일:
        -  이미지 박스 상단 (profile_img_url)
        -  이름, 타이틀, industry/visa 태그 표시
        - 하단 북마크 버튼 + 북마크 수 표시
    - 전체 카드 폭 고정, 태그와 북마크를 같은 라인에 배치
3. 데이터 요청 방식 분기
 - 유저일 때:
    - useHomeJobPosting / useGetCareerList / useGetPostList 사용
    - 북마크, 스크랩 기능 포함
 - 고용주일 때:
    - useInfiniteGetEmployeeResumeList 사용
    - 북마크 추가/삭제 시 usePutScrapResume 사용


## Uncompleted Tasks 😅

[//]: # "없다면 N/A"

- [ ] Task1

## To Reviewers 📢

[//]: # "reviewer가 알면 좋은 내용들"
1. 유저회원
![화면 기록 2025-06-24 오후 9 19 32](https://github.com/user-attachments/assets/3a904bd7-ef84-43a6-be15-df5b8fbe3eb1)
2. 기업회원
![화면 기록 2025-06-24 오후 9 21 29](https://github.com/user-attachments/assets/86804642-4b4a-4381-9389-1f7263a74820)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


## Summary by CodeRabbit

- **New Features**
    - 홈 화면에 커리어 카드 및 커리어 리스트 컴포넌트가 추가되어 커리어 관련 정보를 확인할 수 있습니다.
    - 홈 섹션이 사용자 유형(사업자/일반 사용자)에 따라 트렌딩 이력서, 트렌딩/북마크 채용공고, 커리어 정보를 다르게 보여줍니다.
    - 채용/커리어 카드 리스트, 이력서 카드 리스트에서 타이틀 및 "더보기" 버튼이 지원됩니다.
    - 직원 이력서 카드 컴포넌트와 북마크, 빈 상태 UI 컴포넌트가 새롭게 도입되었습니다.
    - 홈 화면에 개인 맞춤 인사말과 헤드라인을 표시하는 컴포넌트가 추가되었습니다.
    - 홈 배너가 캐러셀 형태로 새롭게 개선되었습니다.
    - 직원 이력서 카드 리스트 컴포넌트가 추가되어 이력서 정보를 가로 스크롤 형태로 제공합니다.
    - 비자 정보 표시를 위한 포맷팅 훅과 관련 유틸리티가 추가되었습니다.

- **개선사항**
    - 이력서 카드 및 리스트, 커리어 카드 리스트의 디자인과 레이아웃이 개선되었습니다.
    - 커리어 검색 섹션이 로그인/비로그인 상태를 구분하여 각각에 맞는 데이터를 보여줍니다.
    - 검색 필터 유틸리티가 사용자/비회원 구분 및 타입 안정성을 강화하도록 개선되었습니다.
    - 홈 메뉴 아이콘과 텍스트가 변경되어 UI가 간결해졌습니다.
    - 채용 카드 및 커리어 카드의 비자 태그 표시 방식이 개선되어 가독성이 향상되었습니다.
    - 일부 컴포넌트의 스타일과 레이아웃이 시맨틱 컬러와 유틸리티 클래스를 활용해 정리되었습니다.
    - 데이터 요청 및 네비게이션 로직이 사용자 유형에 따라 세분화되어 사용자 경험이 향상되었습니다.

- **버그 수정**
    - 일부 리스트에서 데이터가 없거나 로딩 중일 때의 상태 표시가 더 안전하게 처리됩니다.
    - 지역 필터 매핑 시 값이 없을 때도 오류 없이 동작하도록 수정되었습니다.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ohofront committed:
refactor: :recycle: 유저/기업 홈 수정

ohofront committed:
feat: :sparkles: 썸네일 추가

ohofront committed:
refactor: :recycle: 이미지 작아지는 버그 수정

naarang commented on src/components/Home/HomeCareerPostCard.tsx:
```diff
@@ -0,0 +1,64 @@
+  const goToCareerDetailPage = () => {
+    updateCurrentPostId(careerData.id);
+    navigate(`/career/${careerData.id}`);
+  };
```
아마 저희 이제 `updateCurrentPostId(careerData.id);`는 사용하지 않고 `navigate(`/career/${careerData.id}`);`만 사용해도 어떤 공고인지 식별할 수 있을 것입니다!

naarang commented on src/hooks/api/useCareer.ts:
```diff
@@ -186,3 +186,29 @@ export const usePutCareerBookmark = () => {
+    enabled: isEnabled,
+    retry: 1,
+  });
+};
```
주석에서 14.7은 어드민 커리어 리스트 조회 api인데 14.3을 잘못 작성하신 것 같습니다!

naarang commented on src/hooks/api/useCareer.ts:
```diff
@@ -186,3 +186,29 @@ export const usePutCareerBookmark = () => {
+    enabled: isEnabled,
+    retry: 1,
+  });
+};
```
해당 부분 주석도 14.4가 아니라 14.1인 것 같아요!

naarang commented on src/pages/Resume/ScrappedJobPostsPage.tsx:
```diff
@@ -106,7 +109,8 @@ const ScrappedJobPostsPage = () => {
     size: 5,
-    sorting: 'RECENT',
+    sorting: 'POPULAR',
+    category: 'SOME_DEFAULT_CATEGORY', // 🔥 서버 필수값 채워주세요!
```
제가 파악하지 못한 변경사항일수도 있어서 질문드립니다! Category값에 'SOME_DEFAULT_CATEGORY'가 들어가는 것이 맞나요?? api 명세서 상에는 `ACTIVITY, PROGRAM, CONTEST, CLUB`값만 들어갈 수 있도록 되어 있는 것으로 알고 있습니다!

naarang commented on src/components/Home/HomeJobPostingSection.tsx:
```diff
@@ -1,68 +1,158 @@
+            isLoading={bookmarkedCareerLoading}
+            onSeeMoreClick={() =>
+              navigate('/resume/scrapped', { state: { filter: 'Career' } })
+            }
```
로그인하지 않은 게스트 상태에서는 북마크 항목이 안보이는 것으로 알고 있습니다! 코드상으로는 북마크한 항목이 게스트인 경우에도 보여질 것 같은데 확인부탁드립니다~!

naarang reviewed (changes requested):
변경사항이 많았는데 고생많으셨습니다🙇‍♀️ 
올려주신 Gif 파일에서 기업회원의 경우에 "see more"와 "더보기"가 혼용되어 나타나는데 "더보기"로 통일되어야 합니다!

naarang reviewed (changes requested):
그리고 기업 회원 화면에서 인재 카드에서 북마크 버튼이 피그마에 정의된 것보다 큰 것 같은데 확인 부탁드립니다!

또한 인재 카드의 경우에 이름은 한 줄, 자기소개는 2줄이고 넘어가면 ... 처리인 것으로 피그마에 되어있어서 반영해주시면 좋을 것 같습니다!

<img src="https://github.com/user-attachments/assets/f5de0d91-f2a3-4cd4-a194-02fe8f8785cb" width="300" />

naarang commented on src/components/Home/HomeJobPostingSection.tsx:
```diff
@@ -1,68 +1,158 @@
+        resumeSortType: POST_SORTING.POPULAR,
+      },
+    });
   };
```
기업회원인 경우에 "요즘 인기있는 인재" 항목의 더보기 버튼을 클릭하며 지금 '/search'로 이동하는데 커리어 검색으로 이동하는 것이 아니라 '/employer/search' 인 인재검색 페이지로 이동해야 합니다

naarang commented on src/components/PostSearch/CareerSearchSection.tsx:
```diff
@@ -34,40 +37,56 @@ const CareerSearchSection = ({
+  // 게스트 요청
   const {
     data: guestCareerData,
     fetchNextPage: guestFetchNextPage,
-    hasNextPage: guesthasNextPage,
+    hasNextPage: guestHasNextPage,
     isFetchingNextPage: guestIsFetchingNextPage,
     isLoading: guestIsInitialLoading,
   } = useInfiniteGetCareerGuestList(
-    formatCareerSearchFilter(searchOption),
-    !account_type ? true : false,
+    formatCareerSearchFilterForGuest({
+      page: 1,
+      size: 5,
+      searchText: searchOption.searchText,
+      careerSortType: searchOption.careerSortType,
+      careerCategory: searchOption.careerCategory,
+    }),
+    !isLogin,
```
기존에는 isEnabled 값으로 `!account_type ? true : false`로 되어있었는데 `!isLogin`으로 바뀌면서 고용주인 경우에도 게스트 권한으로 커리어 조회가 되는 것 같아서 막아주셔야할 것 같습니다!

naarang commented on src/components/Home/HomeJobPostingSection.tsx:
```diff
@@ -1,68 +1,158 @@
+            data={trendingJobData?.data?.job_posting_list}
+            isLoading={trendingJobLoading}
+            onSeeMoreClick={goToJobSearch}
+          />
```
기업회원 화면에서 "🔥 요즘 인기있는 인재"와 "🔥 요즘 인기있는 공고" 사이에 간격이 피그마보다 큰 것 같은데 확인 부탁드립니다.

<img src="https://github.com/user-attachments/assets/f3ea9581-be71-4719-9a67-b3439b3b02fb" width="300" />

naarang reviewed (changes requested):
일반 회원의 경우에 퀵 메뉴의 메뉴명들도 변경되었습니다! 다음의 피그마를 참고해서 반영해주시면 좋을 것 같습니다~

![image](https://github.com/user-attachments/assets/c2d649b3-6a01-4532-903b-31b5ccf7ef6b)

MrMirror21 committed:
Merge branch 'dev' of https://github.com/Team-inglo/Giggle-Web into feat/413

MrMirror21 committed:
🎨 style: 홈 페이지 내 횡스크롤형 ui 화면 내 padding 적용 방식 수정 #413

- 기존 page 최상위에서 일괄적으로 px-4를 가지고 있었지만 횡스크롤 형 ui들은 px-4에 의해 잘리면 안되므로 해당 ui가 자체적으로 padding을 가지도록 변경함

MrMirror21 committed:
🐛 fix: merge 과정 중 오병합된 코드 롤백 #413

MrMirror21 committed:
🐛 fix: 퀵메뉴 컨테이너의 기본 padding 제거 #413

- 사용자 유형에 따라 퀵메뉴가 존재하지 않음에도 그 container의 padding이 존재, 화면의 불필요한 공백 생성하고 있어 이를 제거함.

MrMirror21 committed:
🎨 style: 홈 페이지 내 ui 레이아웃 변경 #413

MrMirror21 committed:
🎨 style: 커리어, 공고 카드 내 비자 표기 방식 변경 #413

MrMirror21 committed:
♻️ refactor: HomeBanner 내 컴포넌트 분리 #413

- HomePage 내 UI 레이아웃 변경에 따라 인사 문구 등을 출력하는 HomeGreetingSection, 이력서 완성도를 보여주는ResumeHelperBanner, 사용자 유형별 퀵메뉴를 제공하는 HomeMenu를 묶어 HomeTopSection 컴포넌트로 제공
- 기존 광고 배너 캐러셀을 제공했던 HomeBanner를 HomeBannerCarousel로 변경, HomePage에서 단독 호출

MrMirror21 committed:
♻️ refactor: 공고/커리어 카드 내 비자 표기 로직 훅 분리 #413

- 사용자 유형별로 공고/커리어 카드 내 요구 비자 리스트 문구를 출력하는 로직을 커스텀 훅으로 분리
- 로직 내 삼항 연산자 제거

MrMirror21 committed:
♻️ refactor: 공고 검색 페이지 내 검색 옵션 메모이제이션 적용 #413

MrMirror21 committed:
🐛 fix: 홈 피드용 인재 카드와 인재 검색용 인재 카드 분리 #413

MrMirror21 committed:
♻️ refactor: 홈/인재 검색 인재 카드 통합 #413

- 로직과 ui 로직 상 중복되는 부분이 많은 홈/인재 검색에 사용되는 HomeEmployerEmployeeCardList와 EmployerEmployeeCardList에 사용되는 코드를 EmployeeCard로 통합했습니다.

MrMirror21 committed:
🎨 style: 홈 퀵메뉴 아이콘 변경 #413

MrMirror21 committed:
🐛 fix: 요즘 뜨는 인재 더보기 선택 시 이동 페이지 수정 #413

MrMirror21 committed:
🔧 chore: 리뷰 피드백 반영 #413

- 커리어 리스트 요청시 기본 카테고리값 설정
- 커리어 api 훅 주석 수정
- 불필요한 store 조작 코드 제거
